crates/nostr-relay-pool/src/relay/inner.rs

```diff
@@ -1062,8 +1072,12 @@ impl InnerRelay {
                 subscription_id,
                 event,
             } => {
-                self.handle_event_msg(subscription_id.into_owned(), event.into_owned())
-                    .await
+                if self.subscription_exists(&subscription_id).await {
```

Comment:
Instead of checking it here, I would move it to the `handle_event_msg` method and return an error instead (the error is logged in `handle_relay_message`). What do you think?

Something like this:

```rust
// Check tags limit
// ...

// Check if the subscription exists
if !self.subscription_exists(&subscription_id).await {
    return Err(Error::SubscriptionNotFound);
}

// Check if the event is expired 
if event.is_expired() {
    return Err(Error::EventExpired);
}
```

I think there is no need to include the subscription ID or other details, since in the log is included the original message.